src/linear.ts

```diff
@@ -210,18 +210,44 @@ export const linearQF = (
 
   if (options.matchingCapAmount !== undefined && totalCapOverflow > 0n) {
     // redistribute the totalCapOverflow to all
-    for (const recipient in calculations) {
-      const recipientOverflow = calculations[recipient].capOverflow;
-      if (recipientOverflow < 0n) {
-        // recipientOverflow is negative so we can distribute something
-        // to the current recipient
-        // matched : totalMatchedFromUncapped = x : totalCapOverflow
-        const additionalMatch =
-          (calculations[recipient].matched * totalCapOverflow) /
-          totalMatchedFromUncapped;
-        const newMatch = calculations[recipient].matched + additionalMatch;
-        calculations[recipient].matched = newMatch;
+
+    let totalLeft = totalCapOverflow;
+
+    while (totalLeft > 0n) {
```

Comment:
If my understanding is correct, the previous version would distribute the overflow in a single iteration. however, this updated version introduces the while loop which iteratively distributes the remaining overflow until it has been completely exhausted. 
I think this seems to be a more accurate approach as it ensures all overflow is accounted for. I'm curious if this iterative approach impacts the accuracy and fairness of the distribution compared to the single iteration method.
As for the code - I do not see any issues! I think we can come up with better docs as considering the fundamental importance of this process. 

Great work in identifying and addressing this issue! Gonna think on this!